crates/subspace-core-primitives/src/pieces.rs

```diff
@@ -68,74 +70,55 @@ impl AsMut<[u8]> for RecordWitness {
 /// Internally piece contains a record and corresponding witness that together with records root of
 /// the segment this piece belongs to can be used to verify that a piece belongs to the actual
 /// archival history of the blockchain.
-#[derive(Debug, Clone, PartialEq, Eq, Ord, PartialOrd, Hash, Encode, Decode, TypeInfo)]
+#[derive(Debug, Default, Clone, PartialEq, Eq, Ord, PartialOrd, Hash, Encode, Decode, TypeInfo)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-pub struct Piece(#[cfg_attr(feature = "serde", serde(with = "hex::serde"))] Vec<u8>);
-
-impl Default for Piece {
-    fn default() -> Self {
-        Self(vec![0u8; PIECE_SIZE])
-    }
-}
+pub struct Piece(Box<PieceArray>);
 
 impl From<Piece> for Vec<u8> {
     fn from(piece: Piece) -> Self {
-        piece.0
+        piece.0.to_vec()
```

Comment:
This is an array, not a slice. Method doesn't seem to exist here. I could likely work around with `unsafe`, but the whole point of this exercise was to avoid it.